repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -686,15 +687,18 @@ def perform():
             storage_info=indata.storage_info,
             xfs_info=indata.xfs_info) as overlay:
         with overlay.nspawn() as context:
-            target_repoids = _gather_target_repositories(context, indata, prod_cert_path)
-            _create_target_userspace(context, indata.packages, indata.files, target_repoids)
-            # TODO: this is tmp solution as proper one needs significant refactoring
-            target_repo_facts = repofileutils.get_parsed_repofiles(context)
-            api.produce(TMPTargetRepositoriesFacts(repositories=target_repo_facts))
-            # ## TODO ends here
-            api.produce(UsedTargetRepositories(
-                repos=[UsedTargetRepository(repoid=repo) for repo in target_repoids]))
-            api.produce(TargetUserSpaceInfo(
-                path=_get_target_userspace(),
-                scratch=constants.SCRATCH_DIR,
-                mounts=constants.MOUNTS_DIR))
+            # Mount the ISO into the scratch container
+            target_iso = next(api.consume(TargetOSInstallationImage), None)
+            with mounting.mount_upgrade_iso_to_root_dir(overlay.target, target_iso):
+                target_repoids = _gather_target_repositories(context, indata, prod_cert_path)
+                _create_target_userspace(context, indata.packages, indata.files, target_repoids)
+                # TODO: this is tmp solution as proper one needs significant refactoring
```

Comment:
Just curious - what refactoring has been meant by this TODO?

Reply:
@fernflower  the produce of `TMPTargetRepositiesFacts`